nbeats_keras/model.py

```diff
@@ -1,53 +1,93 @@
+        
         self.stack_types = stack_types
         self.nb_blocks_per_stack = nb_blocks_per_stack
-        self.share_weights_in_stack = share_weights_in_stack
         self.thetas_dim = thetas_dim
-        self.best_perf = 100.0
-        self.steps = 10001
-        self.plot_results = 100
+        self.units = hidden_layer_units
+        self.share_weights_in_stack = share_weights_in_stack
+        self.backcast_length = backcast_length
+        self.forecast_length = forecast_length
+        self.best_perf = best_perf
+        self.steps = steps
+        self.plot_results = plot_results
+        self.learning_rate = learning_rate
+        self.loss = loss
+        self.input_dim = input_dim
+        self.exo_dim = exo_dim 
+        self.input_shape = (self.backcast_length, self.input_dim)
+        self.exo_shape = (self.backcast_length, self.exo_dim)
+        self.output_shape = (self.forecast_length, self.input_dim)
         self.weights = {}
         assert len(self.stack_types) == len(self.thetas_dim)
 
-        X_ = Input(shape=(self.backcast_length,))
-        x_ = Lambda(lambda x: x)(X_)
+        X = Input(shape=self.input_shape)
+        x_ = {}
+        for l in range(self.input_dim):
+            x_[l] = Lambda(lambda x: x[:, :, l])(X)
+        e_ = {}
+        if self.exo_dim > 0:
+            E = Input(shape=self.exo_shape)
+            for l in range(self.exo_dim):
+                e_[l] = Lambda(lambda x: x[:, :, l])(E)       
+        y_ = {}
+        
         for stack_id in range(len(self.stack_types)):
             stack_type = self.stack_types[stack_id]
             nb_poly = self.thetas_dim[stack_id]
             for block_id in range(self.nb_blocks_per_stack):
-                backcast, forecast = self.create_block(x_, stack_id, block_id, stack_type, nb_poly)
-                x_ = Subtract()([x_, backcast])
-                if stack_id == 0 and block_id == 0:
-                    y_ = forecast
-                else:
-                    y_ = Add()([y_, forecast])
-
-        model = Model(X_, y_)
+                backcast, forecast = self.create_block(x_, e_, stack_id, block_id, stack_type, nb_poly)
+                for l in range(self.input_dim):
+                    x_[l] = Subtract()([x_[l], backcast[l]])                    
+                    if stack_id == 0 and block_id == 0:
+                        y_[l] = forecast[l]
+                    else :
+                        y_[l] = Add()([y_[l], forecast[l]])
+        
+        for l in range(self.input_dim):
+            y_[l] = Reshape(target_shape=(self.forecast_length, 1))(y_[l])
+        if self.input_dim > 1:
+            y_ = Concatenate(axis=-1)(list(y_.values()))
+        else:
+            y_ = y_[0]
+            
+        if self.exo_dim > 0:
+            model = Model([X, E], y_)
+        else:
+            model = Model(X, y_)
+            
         model.summary()
-
+        
         self.nbeats = model
+        self.compile_model()
```

Comment:
Don't need to compile model by default I guess.

Reply:
Updated